types/ramda/index.d.ts

```diff
@@ -145,9 +145,9 @@ declare namespace R {
         (t1: T1, t2: T2, t3: T3, t4: T4, t5: T5, t6: T6): R;
     }
 
-    interface Reduced {
-        [index: number]: any;
-        [index: string]: any;
+    interface Reduced<T> {
+        '@@transducer/value': T;
```

Comment:
I don't understand these properties - are they really part of the API? Can you provide a doc reference please?

Reply:
Seems to me like they are:
https://github.com/ramda/ramda/blob/master/source/internal/_reduced.js

Reply:
Yes they are. They are an implementation details that is subject to change, but I still think we should include this in the type definitions. Unfortunately there is no way in JavaScript or TypeScript to enforce encapsulation without using classes, AFAIK. And I don't think it is the job of the TypeScript definitions to enforce that kind of thing either, it should just represent the actual types used in the JavaScript library.

Reply:
> And I don't think it is the job of the TypeScript definitions to enforce that kind of thing either, it should just represent the actual types used in the JavaScript library.

This is up to interpretation. Of course the typings should represent actual types, but the goal of the typings is to allow developers to easily write correct, maintainable code. More often than not, this involves excluding implementation details and undocumented properties.

I'm not familiar with this particular library, so I'm going to need your help understanding the intention. To demonstrate, could you add a test case, showing a common use case, that would fail without your changes.

Reply:
Good idea, I will see if I can add a test case if I have the time